### PR TITLE
Rename UC_OSX (and related constants) to UC_MAC

### PR DESCRIPTION
**Pull request by vomindoraan**
## Description

Apple moved away from calling their desktop operating system OS X in 2016, and macOS is now the preferred name. We've been following this convention in docs and code for the most part, and I think now is a good time to rename the Unicode constants as well.

This PR renames the following constants:

|Old name          |New name          |Type                |
|------------------|------------------|--------------------|
|`UC_OSX`          |`UC_MAC`          |Input mode name     |
|`UNICODE_MODE_OSX`|`UNICODE_MODE_MAC`|Keycode name        |
|`UC_M_OS`         |`UC_M_MA`         |Keycode alias       |
|`UNICODE_KEY_OSX` |`UNICODE_KEY_MAC` |Overridable constant|
|`UNICODE_SONG_OSX`|`UNICODE_SONG_MAC`|Overridable constant|

The old names are retained as aliases for compatibility reasons, but have been marked as deprecated in code and docs.

Note: This PR is a reboot of #6401.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by vomindoraan** [comment]
I'd rather avoid the breaking change route so this doesn't end up dead in the water again. Let me know if I should revert the keymap changes and add legacy aliases for the other constants as well in order to avoid that.

The other option is to go full-on breaking change mode and replace all usages of the constants in the codebase with new ones.

File: quantum/process_keycode/process_unicode_common.h [hunk 1]
```diff
@@ -23,8 +23,8 @@
 #endif
 
 // Keycodes used for starting Unicode input on different platforms
-#ifndef UNICODE_KEY_OSX
-#    define UNICODE_KEY_OSX KC_LALT
+#ifndef UNICODE_KEY_MAC
+#    define UNICODE_KEY_MAC KC_LALT
```
Comment (skullydazed) [inline quantum/process_keycode/process_unicode_common.h]:
```suggestion
#ifndef UNICODE_KEY_MAC
#    define UNICODE_KEY_OSX KC_LALT
#    define UNICODE_KEY_MAC KC_LALT
```

File: quantum/quantum_keycodes.h [hunk 2]
```diff
@@ -777,7 +777,8 @@ enum quantum_keycodes {
 #define UC_MOD UNICODE_MODE_FORWARD
 #define UC_RMOD UNICODE_MODE_REVERSE
 
-#define UC_M_OS UNICODE_MODE_OSX
+#define UC_M_MA UNICODE_MODE_MAC
+#define UC_M_OS UNICODE_MODE_MAC // Deprecated alias
```
Comment (skullydazed) [inline quantum/quantum_keycodes.h]:
```suggestion
#define UNICODE_MODE_OSX UNICODE_MODE_MAC
#define UC_M_MA UNICODE_MODE_MAC
#define UC_M_OS UNICODE_MODE_MAC // Deprecated alias
```

**Comment by Erovia** [comment]
I'm okay with this, if we leave the old names working.  
Breaking things just to correct something only some lawyers and marketing teams care about in Cupertino is not something I can swallow.

tl;dr: allow the usage of correct names besides the historical ones

**Comment by vomindoraan** [comment]
I've added backwards-compatible aliases for the existing names, as requested. I've reverted the changes made to user keymaps, but I opted to keep the updated names in Drashna and kuchosauronad0's userspaces since they have signed off on the changes.

**Comment by vomindoraan** [comment]
I believe this can be unmarked as a breaking change now.
